lib/api/core/plugins/pluginsManager.js

```diff
@@ -150,22 +161,19 @@ class PluginsManager {
           initTimeout
         } = this.config.common;
 
-      if (plugin.config.threads) {
+      if (plugin.object.threadable && plugin.config.threads) {
         debug('[%s] starting worker with %d threads', pluginName, plugin.config.threads);
 
         return initWorkers(plugin, pluginName, this.kuzzle.config)
           .timeout(initTimeout)
-          .catch(err => {
-            // eslint-disable-next-line no-console
-            this.silent || console.log('Plugin', pluginName, ' errored with message: ', err.message, '. Skipping...');
-            return Bluebird.resolve();
-          });
+          .catch(err => Bluebird.reject(new PluginImplementationError(`Unable to start worker plugin "${pluginName}"; ${err.message}`)));
       }
 
+
       debug('[%s] starting plugin in "%s" mode', pluginName, plugin.config.privileged ? 'privileged' : 'standard');
 
       // We sandbox the Promise.resolve into a Promise to be able to catch throwed errors if any
-      return new Bluebird(resolve => {
+      return new Bluebird((resolve, reject) => {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ASemicolon)